src/main/scala/com/cognite/sdk/scala/v1/resources/dataPoints.scala

```diff
@@ -74,21 +162,37 @@ class DataPointsResourceV1[F[_]](val requestSession: RequestSession[F])
 
   def queryById(id: Long, inclusiveStart: Long, exclusiveEnd: Long): F[Seq[DataPoint]] =
     requestSession
-      .sendCdf { request =>
+      .postJSONAcceptProtobuf { request =>
         request
           .post(uri"$baseUri/list")
           .body(Items(Seq(QueryRangeById(id, inclusiveStart.toString, exclusiveEnd.toString))))
-          .response(asJson[Either[CdpApiError, Items[DataPointsByIdResponse]]])
+          .response(asProtobufOrCdpApiError)
           .mapResponse {
             case Left(value) => throw value.error
-            case Right(Left(cdpApiError)) => throw cdpApiError.asException(uri"$baseUri/list")
-            case Right(Right(value)) =>
-              value.items.headOption match {
-                case Some(items) => items.datapoints
-                case None => Seq.empty
-              }
+            case Right(Left(cdpApiError)) => throw cdpApiError.asException(uri"$baseUri/delete")
+            case Right(Right(dataPointListResponse)) =>
+              parseNumericDataPoints(dataPointListResponse)
+          }
+      }
+
+  val asProtobufOrCdpApiError: ResponseAs[
+    Either[DeserializationError[io.circe.Error], Either[CdpApiError, DataPointListResponse]],
+    Nothing
+  ] = {
+    asByteArray.map(response => {
+      try {
+        Right(Right(DataPointListResponse.parseFrom(response)))
+      } catch {
+        case _: Throwable =>
```

Comment:
We can leave it as a "TODO" comment, but we shouldn't forget that we can/should use the HTTP headers to attempt the correct parsing directly.